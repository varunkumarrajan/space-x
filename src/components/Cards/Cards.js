// React
import React from 'react';
// Prop types
import PropTypes from 'prop-types'; // ES6
// Styled component
import styled from 'styled-components';
// Loader
import BlockUi from 'react-block-ui';
// Components
import Card from 'components/Card/Card';

// Styled card wrapper
const CardsWrapper = styled.section`
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
  flex: 0 0 calc(100% - 280px);
`;

// Styled heading wrapper
const HeadingWrapper = styled.h2`
  margin: 10px;
`;

/**
 * @description cards component return the array of card
 * @param {object} { content }
 */
const Cards = ({ content }) => (
  <BlockUi tag='div' blocking={content.loading}>
    <CardsWrapper>
      {content.launchDetails.length > 0 ? (
        content.launchDetails.map((lan, ind) => (
          <Card
            key={ind}
            content={{
              labels: content.labels,
              alt: lan?.details ?? '',
              src: lan?.links?.mission_patch ?? '',
              missionName: `${lan.mission_name} # ${lan.flight_number}`,
              missionIds: lan.mission_id,
              year: lan.launch_year,
              sLaunch: lan?.launch_success?.toString() ?? '',
              sLand: lan?.launch_landing?.toString() ?? '',
            }}
          />
        ))
      ) : (
        <React.Fragment>
          {!content.loading && (
            <HeadingWrapper>{content.noContent}</HeadingWrapper>
          )}
        </React.Fragment>
      )}
    </CardsWrapper>
  </BlockUi>
);

// Default export
export default Cards;

// Default props
Cards.defaultProps = {
  content: {
    noContent: '',
    labels: {},
    loading: false,
    launchDetails: [],
  },
};

// Prop types
Cards.propTypes = {
  content: PropTypes.shape({
    noContent: PropTypes.string.isRequired,
    labels: PropTypes.any.isRequired,
    loading: PropTypes.bool.isRequired,
    launchDetails: PropTypes.arrayOf(PropTypes.any).isRequired,
  }).isRequired,
};
