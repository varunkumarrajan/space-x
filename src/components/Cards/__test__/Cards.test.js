// React
import React from 'react';
// Testing library
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Cards from './../Cards';

// Mock data
const cardsMock = {
  noContent: 'testing',
  labels: {},
  loading: false,
  launchDetails: [],
}

afterEach(cleanup);

describe('Cards', () => {
  it('Should Render Cards With Props', () => {
    const { container } = render(<Cards content={cardsMock} />);
    expect(container.children).not.toHaveLength(0);
  });

  it('Cards Snapshot', () => {
    const cards = renderer.create(<Cards content={cardsMock} />).toJSON();
    expect(cards).toMatchSnapshot();
  });
});
