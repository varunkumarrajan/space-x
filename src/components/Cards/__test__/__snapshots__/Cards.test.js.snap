// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Cards Snapshot 1`] = `
<div
  aria-busy={false}
>
  <section
    className="sc-fubCfw jetgdW"
  >
    <h2
      className="sc-pFZIQ bbSUyC"
    >
      testing
    </h2>
  </section>
  <span />
</div>
`;
